tests/data/attributes/test_attribute.py

```diff
@@ -6,11 +6,12 @@
     DatasetNone,
     DatasetScalar,
     DatasetString,
+    DatasetSparseArray,
     DatasetOperator,
 )
 from pennylane.data.attributes import DatasetArray, DatasetScalar, DatasetString, DatasetNone
 from pennylane.data.base.attribute import match_obj_type
-import pennylane as qml
+from pennylane.data.attributes.sparse_array import _ALL_SPARSE
```

Comment:
Perhaps this is an indication that `_ALL_SPARSE` should not be listed as private.